src/go-camo/camo/proxy.go

```diff
@@ -61,6 +73,104 @@ type Proxy struct {
 	config    *Config
 }
 
+func isContentTypeAcceptable(contentType string) bool {
+	for _, prefix := range acceptedContentTypes {
+		if strings.HasPrefix(contentType, prefix) {
+			return true
+		}
+	}
+	return false
+}
+
+func hexEncodeCSSURLs(baseURL *url.URL, hmacKey []byte, css []byte) ([]byte, error) {
+	re, err := regexp.Compile(`(?:url[\s]?)(?:\(['"]?)(.*?)(?:['"]?\))`)
+	if err != nil {
+		return nil, err
+	}
+
+	return re.ReplaceAllFunc(css, func(b []byte) []byte {
+		cssURL := re.FindSubmatch(b)[1]
+		if cssURL == nil {
+			return b
+		}
+
+		// Convert to absolute URL
+		u, err := url.Parse(string(cssURL))
+		if err != nil {
+			return b
+		}
+		u = baseURL.ResolveReference(u)
+
+		// Encode to proxy URL
+		hexURL := encoding.HexEncodeURL(hmacKey, u.String())
+		return bytes.Replace(b, cssURL, []byte(hexURL), -1)
+	}), nil
+}
+
+func writeCSSWithResolvedURLs(baseURL *url.URL, contentEncoding string, hmacKey []byte, w io.Writer, r io.ReadCloser) (int64, error) {
+	var err error
+	if contentEncoding == "gzip" {
+		r, err = gzip.NewReader(r)
+		if r != nil {
+			defer r.Close()
+		}
+		if err != nil {
+			return 0, err
+		}
+	}
+
+	css, err := ioutil.ReadAll(r)
+	if err != nil {
+		return 0, err
+	}
+
+	resolvedCSS, err := hexEncodeCSSURLs(baseURL, hmacKey, css)
+	if err != nil {
+		return 0, err
+	}
+
+	errChan := make(chan error)
+	successChan := make(chan bool)
+
+	var pr io.Reader
+
+	if contentEncoding == "gzip" {
+		var pw io.WriteCloser
+		pr, pw = io.Pipe()
+
+		go func() {
+			defer pw.Close()
+
+			gz := gzip.NewWriter(pw)
+			if gz != nil {
+				defer gz.Close()
+			}
+			if _, err := gz.Write(resolvedCSS); err != nil {
+				errChan <- err
+			}
+			close(successChan)
+		}()
+	} else {
+		pr = bytes.NewReader(resolvedCSS)
+		close(successChan)
+	}
+
+	n, err := io.Copy(w, pr)
+	if err != nil {
+		return 0, err
+	}
+
+	select {
+	case err := <-errChan:
+		return 0, err
+	case <-successChan:
+	case <-time.After(time.Second * 60):
```

Comment:
:+1: Nice defensive programming. Not sure about what considerations you have made surrounding this, but I would think that you would want to make this configurable, or at least shorter than 60 seconds.

Reply:
Thanks 😄 I didn't want to potentially introduce any _deadlocks_. Though, with `n, err := io.Copy(w, pr)`, technically a response is received by the client before this function even properly _returns_! But just to be sure, we should timeout anyway. 

And yes, it really should be <10s. No writes should ever take that long! And if it does, it is probably going to throw an error beforehand. I have no intentions on making it configurable since this seems like an implementation detail. It is not something an end-user would necessarily care about.